tests/api/runtimes/test_kubejob.py

```diff
@@ -141,6 +147,275 @@ def test_run_with_node_selection(self, db: Session, client: TestClient):
+        mlrun.mlconf.preemptible_nodes.node_selector = base64.b64encode(
+            json.dumps(node_selector).encode("utf-8")
+        )
+        runtime = self._generate_runtime()
+        runtime.with_preemption_mode(PreemptionModes.prevent)
+        self._execute_run(runtime)
+
+        self.assert_node_selection(affinity=self._generate_preemptible_anti_affinity())
+
+        # tolerations are set, but expect to stay because those tolerations aren't configured as preemptible tolerations
+        runtime = self._generate_runtime()
+        runtime.with_node_selection(tolerations=self._generate_tolerations())
+        runtime.with_preemption_mode(PreemptionModes.prevent)
+        self._execute_run(runtime)
+        self.assert_node_selection(
+            affinity=self._generate_preemptible_anti_affinity(),
+            tolerations=self._generate_tolerations(),
+        )
+
+        # set default preemptible tolerations
+        tolerations = self._generate_tolerations()
+        serialized_tolerations = self.k8s_api.sanitize_for_serialization(tolerations)
+        mlrun.mlconf.preemptible_nodes.tolerations = base64.b64encode(
+            json.dumps(serialized_tolerations).encode("utf-8")
+        )
+        # tolerations are set, expect preemptible tolerations to be removed and anti-affinity not to be set
+        runtime = self._generate_runtime()
+        runtime.with_node_selection(
+            tolerations=self._generate_preemptible_tolerations()
+        )
+        runtime.with_preemption_mode(PreemptionModes.prevent)
+        self._execute_run(runtime)
+        self.assert_node_selection(affinity=self._generate_preemptible_anti_affinity())
+
+    def test_run_with_constrain_preemptible_mode(self, db: Session, client: TestClient):
+        node_selector = self._generate_node_selector()
+        mlrun.mlconf.preemptible_nodes.node_selector = base64.b64encode(
+            json.dumps(node_selector).encode("utf-8")
+        )
+        runtime = self._generate_runtime()
+        runtime.with_preemption_mode(PreemptionModes.constrain)
+        self._execute_run(runtime)
+        self.assert_node_selection(affinity=self._generate_preemptible_affinity())
+        # set default preemptible tolerations
+        tolerations = self._generate_tolerations()
+        serialized_tolerations = self.k8s_api.sanitize_for_serialization(tolerations)
+        mlrun.mlconf.preemptible_nodes.tolerations = base64.b64encode(
+            json.dumps(serialized_tolerations).encode("utf-8")
+        )
+        runtime = self._generate_runtime()
+        runtime.with_preemption_mode(PreemptionModes.constrain)
+        self._execute_run(runtime)
+        self.assert_node_selection(
+            affinity=self._generate_preemptible_affinity(),
+            tolerations=self._generate_preemptible_tolerations(),
+        )
+        # sets different affinity before, expects to override the required_during_scheduling_ignored_during_execution
```

Comment:
Why override is expected and not merging ?

Reply:
When using constrain we want the function to run **only** on preemptible nodes, if we merge both existing and preemptible affinity, this will result the function to be able to run not only on preemptible nodes.